nimble_builder/__init__.py

```diff
@@ -1,3 +1,153 @@
+        raise ValueError(f"Unknown screw size {self.mounting_screws}")
+
+    @property
+    def mounting_hole_tap_diameter(self):
+        """
+        Return the diameter for a tapped hole for the front mounting screws.
+        This is determined by the `mounting_screws` parameter.
+        Tap holes should be in the legs where the screw will tap.
+        As 3D printers tend to over extrude ad the fact that the machine screws
+        are tapping directly into plastic the holes are larger than standard
+        metric tap holes.
+        See also: mounting_hole_tap_diameter
+        """
+        if self.mounting_screws == "M4":
+            return 3.6
+        if self.mounting_screws == "M6":
+            return 5.5
+        raise ValueError(f"Unknown screw size {self.mounting_screws}")
+
+    @property
+    def end_plate_hole_clearance_diameter(self):
+        """
+        Return the diameter for a clearance hole for the countersunk screws
+        that attach the top and bottom plates.
+        This is determined by the `end_plate_screws` parameter.
+        Clearance holes should be used in end_plates.
+        See also: mounting_hole_tap_diameter
+        """
+        if self.end_plate_screws == "M5":
+            return 5.4
+        if self.end_plate_screws == "M6":
+            return 6.5
+        raise ValueError(f"Unknown screw size {self.end_plate_screws}")
+
+    @property
+    def end_plate_hole_countersink_dia(self):
+        """
+        Return the countersink diameter for a clearance hole for the countersunk screws
+        that attach the top and bottom plates.
+        This is determined by the `end_plate_screws` parameter.
+        See also: mounting_hole_tap_diameter
+        """
+        if self.end_plate_screws == "M5":
+            return 11.5
+        if self.end_plate_screws == "M6":
+            return 13.7
+        raise ValueError(f"Unknown screw size {self.end_plate_screws}")
+
+    @property
+    def end_plate_hole_tap_diameter(self):
+        """
+        Return the diameter for a tapped hole for the countersunk screws
+        that attach the top and bottom plates.
+        This is determined by the `end_plate_screws` parameter.
+        Tap holes should be in the top and bottom of the legs where the screw will tap.
+        As 3D printers tend to over extrude ad the fact that the machine screws
+        are tapping directly into plastic the holes are larger than standard
+        metric tap holes.
+        See also: mounting_hole_tap_diameter
+        """
+        if self.end_plate_screws == "M5":
+            return 4.5
+        if self.end_plate_screws == "M6":
+            return 5.5
+        raise ValueError(f"Unknown screw size {self.end_plate_screws}")
+
+    end_plate_hole_dia: float = 4.7
+    end_plate_hole_countersink_dia: float = 11
+
+    @property
+    def rack_width(self):
+        """
+        Return the rack width in mm as determined by the nominal_rack_width.
+        Options are:
```

Comment:
If the docstrings are going to be exported to something like ReadTheDocs, there are tags like `param` that could be used here. This may not be desired/necessary though.

Reply:
Again I agree. Doing this would be good, but would be scope creep for this issue. I have made issue #60 so we don't forget this point and will resolve this conversation